### PR TITLE
OSD: Input recording overlap fixed

### LINKED ISSUE #8330: [BUG]: Input recording hotkey + OSD overlap
### Describe the Bug

- Shift + R in hotkeys doesn't do anything (anymore) to start new input recording
- Input recording overlaps with OSD
image

- Third issue is when you already had made one you can't just pick folder again, you do need to type it below with my example being 'aaaa' and it auto-creates it, so should be made more visible that's not open but create perhaps:
image

### Reproduction Steps

Press Shift + R for first issue

For second issue:
- Run a game
- Then make new recording
image
- Apply multiple OSD options

Third issue:
- Try make new recording
- Requests specific p2m2

### Expected Behavior

All to work fine as expected when I want a new one pick specific file or folder, hotkey to work and also don't destroy readability with OSD.

### PCSX2 Revision

1.7.4187

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

_No response_

### PR DESCRIPTION
**Pull request by sestain**
### Description of Changes
Draws the input recording overlay first and then the performance overlays.

### Rationale behind Changes
Closes #8330.

### Suggested Testing Steps
Check by playing an input recording while all OSD settings are on.

### CODE REVIEW

**Review by Mrlinkwii — APPROVED** [review]
image
 looks good

File: pcsx2/ImGui/ImGuiOverlays.cpp [hunk 1]
```diff
@@ -51,6 +51,8 @@
 #include <tuple>
 #include <unordered_map>
 
+float position_y = 0;
```
Comment (stenzek) [inline pcsx2/ImGui/ImGuiOverlays.cpp:54]: Please don't add additional global variables, polluting the global namespace.

This is also going to break if you turn recording off. Have a local in the parent function, and pass it in by reference.

**Review by stenzek — CHANGES_REQUESTED** [review]

File: pcsx2/ImGui/ImGuiOverlays.cpp [hunk 2]
```diff
@@ -666,8 +665,9 @@ void ImGuiManager::DrawInputRecordingOverlay()
 
 void ImGuiManager::RenderOverlays()
 {
-	DrawPerformanceOverlay();
-	DrawInputRecordingOverlay();
+	float position_y;
```
Comment (stenzek) [inline pcsx2/ImGui/ImGuiOverlays.cpp:668]: Initialize this with zero, then in `DrawInputRecordingOverlay()`, add `margin` to it. That way if something gets added in the middle later, it won't break.

**Comment by kamfretoz** [comment]
This is just my two cents... Instead of cramming the input recording overlay on the top right along with performance stats, why not just put it on the bottom left just above the controllers input indicator? it fits more there imo.

**Comment by refractionpcsx2** [comment]
I don't think adding to the y position is really the solution, what if you stop input recording? it's going to stay offset.

I suggest you do what Stenzek originally suggested and move the assignment off in to a separate function you can call from two places. Also the add of the initial margin probably should be in RenderOverlays(), otherwise if you're not recording, there won't be a margin.

**Comment by sestain** [comment]
> I don't think adding to the y position is really the solution, what if you stop input recording? it's going to stay offset.

What do you mean exactly by "what if you stop input recording? it's going to stay offset."?

**Comment by refractionpcsx2** [comment]
I mean you're offsetting the OSD to make room for the input recording information, then stop the input recording, the OSD is going to still be offset, no?

**Comment by sestain** [comment]
Start of emulation
image
Start of input recording
image
End of input recording by stopping it
image

**Comment by refractionpcsx2** [comment]
Huh, alright then.

**Pull request merged**